robottelo/host_helpers/ui_factory.py

```diff
@@ -0,0 +1,58 @@
+"""
+It is not meant to be used directly, but as part of a robottelo.hosts.Satellite instance
+Need to pass the existing session object to the ui_factory method as a parameter
+example: my_satellite.ui_factory(session).ui_method()
+"""
+from fauxfactory import gen_string
+
+from robottelo.constants import DEFAULT_CV
+from robottelo.constants import ENVIRONMENT
+
+
+class UIFactory:
+    """This class is part of a mixin and not to be used directly. See robottelo.hosts.Satellite"""
+
+    def __init__(self, session):
+        self._session = session
+
```

Comment:
```suggestion
    def __init__(self, satellite):
        self._session = satellite.ui
```
if you have `self.ui` attribute in Satellite class